pex/interpreter_constraints.py

```diff
@@ -134,6 +134,9 @@ def __bool__(self):
         # type: () -> bool
         return bool(self.constraints)
 
+    # N.B.: For Python 2.7.
```

Comment:
When adding `__bool__` to the new `ScriptMetadata` class in `pex/pep_723.py` below, I did a sweep to make sure `__nonzero__` was being used everywhere and this was the 1 missing spot. Otherwise unrelated to this change.